sql/src/main/java/io/crate/executor/transport/TransportFetchNodeAction.java

```diff
@@ -101,7 +101,7 @@ public void nodeOperation(final NodeFetchRequest request,
             statsTables.operationStarted(request.fetchPhaseId(), request.jobId(), "fetch");
 
             // nothing to fetch, just close
-            if (request.toFetch() == null) {
+            if (request.toFetch() == null && request.isCloseContext()) {
```

Comment:
does this mean we send requests where there is nothing to fetch and where isCloseContext is false? Isn't that a bit of a waste ?